test/libsolidity/SyntaxTest.cpp

```diff
@@ -83,7 +83,18 @@ TestCase::TestResult SyntaxTest::run(ostream& _stream, string const& _linePrefix
 		OptimiserSettings::minimal()
 	);
 	if (compiler().parse())
-		compiler().analyze();
+		if (compiler().analyze())
+			try {
+				if (!compiler().compile())
+					BOOST_THROW_EXCEPTION(runtime_error("Compilation failed even though analysis was successful."));
```

Comment:
Maybe, but fine with how it is.
```suggestion
BOOST_THROW_EXCEPTION(runtime_error("Analysis succeeded. Compilation failed."));
```

Reply:
I'd prefer it how it is to emphasize that this isn't something that should ever happen...